File: solutions/fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let toRna (dna: string) : string =
    seq {
        for b in dna do
            yield
                match b with
                | 'G' -> 'C'
                | 'C' -> 'G'
                | 'T' -> 'A'
                | 'A' -> 'U'
                | _ -> failwith "Invalid base"
    }
    |> System.String.Concat

File: solutions/julia/rna-transcription/rna-transcription.jl
dna_to_rna = Dict('G' => 'C',
                  'C' => 'G',
                  'T' => 'A',
                  'A' => 'U')

function to_rna(dna)
    try
        string((dna_to_rna[c] for c in dna)...)
    catch 
        throw(ErrorException("Invalid DNA base"))
    end
end
